qt/widgets/instrumentview/src/InstrumentWidgetPickTab.cpp

```diff
@@ -1237,27 +1190,25 @@ void DetectorPlotController::plotSingle(int detid) {
 *   LENGTH
 *   PHI
 * The units can be set with setTubeXUnits(...) method.
-* @param detid :: A detector id. The miniplot will display data for a component
+* @param detindex :: A detector index. The miniplot will display data for a
+* component
 * containing the detector
 *   with this id.
 */
-void DetectorPlotController::plotTube(int detid) {
+void DetectorPlotController::plotTube(size_t detindex) {
   const auto &actor = m_instrWidget->getInstrumentActor();
-  auto &det = actor.getDetectorByDetID(detid);
-  boost::shared_ptr<const Mantid::Geometry::IComponent> parent =
-      det.getParent();
-  Mantid::Geometry::ICompAssembly_const_sptr ass =
-      boost::dynamic_pointer_cast<const Mantid::Geometry::ICompAssembly>(
-          parent);
-  if (parent && ass) {
+  const auto &componentInfo = actor.getComponentInfo();
+
+  if (componentInfo.hasParent(detindex) &&
+      componentInfo.detectorsInSubtree(detindex).size() > 0) {
```

Comment:
This can be expensive `detectorsInSubtree` as it will involve a large allocation. And you only want to know the size. Maybe consider adding a method to `ComponentInfo` to get you the number of detectors in the subtree. Can be calculated from the ranges.